### PR TITLE
require same version of thrust

### LINKED ISSUE #4: How to deal with the version lock with Thrust?
Hi, @jakirkham told me this feedstock, thanks for making it live. Since newer CUB will be version-locked with Thrust (see the links below), I am a bit worried about using this feedstock and, in particular, whether things can/cannot be overridden. Can NVIDIA folks provide comments and suggestions?

refs:
1. https://github.com/thrust/thrust/pull/1082
2. https://github.com/NVlabs/cub/issues/182#issuecomment-630938376
3. https://github.com/rapidsai/rmm/pull/378#issuecomment-635526868

### PR DESCRIPTION
**Pull request by rongou**
Fixes #4 

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by rongou** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,9 +10,13 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 1
+  number: 2
   script: mkdir -p "$PREFIX"/include && cp -r cub "$PREFIX"/include  # [not win]
 
+requirements:
+  run:
+    - thrust {{ version }}.*
```
Comment (jakirkham) [inline recipe/meta.yaml:18]: Do we intend to make `thrust` a requirement here or merely align versions? If the former, this seems fine. If the latter, we can use [`run_constrained`]( https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#run-constrained ) to ensure that this version constraint is met when the user installs `thrust`, but doesn't actually require `thrust` to be installed.

Comment (leofang) [reply recipe/meta.yaml:18]: Thrust is a dependency of CUB.

Comment (leofang) [reply recipe/meta.yaml:18]: I am curious if we should add conda's compilers here?

**Pull request merged**